### DISCUSSION TITLE
Storing information permanently

### DISCUSSION THREAD

**Discussion by mcc-devel** [Rapptz/discord.py#9180] [Q&A] [answered]
I'm writing a moderation bot for my server and I need to know how many strikes each member has. How do I store that?

**Comment by lightlybakedman** [comment, marked as answer]
You'd store this data inside a database such as sqlite or postgresql. Use async libraries for these though, like asqlite/aiosqlite and asyncpg.

**Reply by mcc-devel** [reply]
Another thing, do I have to store this in a personal server or does discord have an api to store it in their servers?

**Reply by its-truce** [reply]
You'd store them in the database?

**Reply by lightlybakedman** [reply]
> Another thing, do I have to store this in a personal server or does discord have an api to store it in their servers?

What servers? I already said to use a database, this can either be local in a file (sqlite) or one you host (postgres).

**Reply by mcc-devel** [reply]
> > Another thing, do I have to store this in a personal server or does discord have an api to store it in their servers?
> 
> What servers? I already said to use a database, this can either be local in a file (sqlite) or one you host (postgres).

Right but I'm gonna host this in repl.it I'm not gonna buy a 24-7 server

**Reply by IAmTomahawkx** [reply]
Then you aren't storing any information. You need a proper server solution to actually store data. Free "solutions" don't provide such things

**Reply by scrazzz** [reply]
> Right but I'm gonna host this in repl.it I'm not gonna buy a 24-7 server.

Repl.it has an in-built database thingy iirc (don't ask me for help, I have no knowledge about it).
Using a json file to store is also an option.
Or use a cloud-based database like MongoDB.

**Comment by mcc-devel** [comment]
# This has been concluded
Please do not send further messages.
